site/content/en/docs/Guides/Client SDKs/local.md

```diff
@@ -149,3 +172,52 @@ wget https://raw.githubusercontent.com/googleforgames/agones/{{< release-branch
 chmod o+r gameserver.yaml
 docker run --network=host --rm -v $(pwd)/gameserver.yaml:/tmp/gameserver.yaml us-docker.pkg.dev/agones-images/release/agones-sdk:{{<release-version>}} --local -f /tmp/gameserver.yaml
 ```
+
+## Running locally using "out of cluster" mode
+
+An alternative to running completely isolated from a cluster is to run in "out of cluster" mode.
+This allows you to run locally but interact with the controllers within a cluster.
+This workflow works well when running a [Local Game Server](https://agones.dev/site/docs/guides/local-game-server/) in your cluster and want to run the server locally.
+This means being able to allocate your game in its normal flow (much more prod-like environment) and be able to debug (e.g. breakpoint) your server code.
+This can also be done with [running Minikube locally](https://agones.dev/site/docs/installation/creating-cluster/minikube/), which is great for early prototyping.
+
+The name "out of cluster" mode is to contrast [InClusterConfig](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#InClusterConfig) which is used in the internal golang kubeconfig API.
+
+To run in "out of cluster" mode, instead of passing `--local` or `-f ./gameserver.yaml`, you'd use `--kubeconfig` to connect into your cluster.
+However, there are a number of commands that are necessary/useful to run when running in "out of cluster" mode.
+Here is a sample with each piece discussed after.
+```bash
+GAMESERVER_NAME='my-local-server' \
```

Comment:
I feel like to do this cleanly, we should update the sdkserver to also allow command line args for the GameServer name and Pod Namespace -- just to be consistent, and make life easier for end users. Then we don't have to mix args and env vars.

Also would prefer to see these docs using a binary, not running from source.

Reply:
> Then we don't have to mix args and env vars.

I agree that this would be nicer. I can add in the extra flags.

> Also would prefer to see these docs using a binary, not running from source.

As mentioned in the other comment, I think running from source is nicer than downloaded binaries. But sure, for the sake of documentation consistency, I can update this.

Reply:
Just to follow up here, this work has been done. `--gameserver-name` and `--pod-namespace` can be used as cli args, and `GAMESERVER_NAME` and `POD_NAMESPACE` still work as env vars (which are used in k8s). Also updated this to be consistent and use pre-built binary commands.